folksonomy/api.py

```diff
@@ -321,20 +322,26 @@ async def product_tag_add(response: Response,
     """
 
     check_owner_user(user, product_tag.owner, allow_anonymous=False)
-    try:
-        timing = await db_exec("""
-INSERT INTO folksonomy (product,k,v,owner,version,editor,comment)
-    VALUES (%s,%s,%s,%s,%s,%s,%s)
-    """, (product_tag.product, product_tag.k.lower(), product_tag.v, product_tag.owner,
-            product_tag.version, user, product_tag.comment
-          ))
-    except psycopg2.Error as e:
-        error_msg = re.sub(r'.*@@ (.*) @@\n.*$', r'\1', e.pgerror)[:-1]
-        return JSONResponse(status_code=422, content={"detail": {"msg": error_msg}})
+    if product_tag.version != 1:
+        raise HTTPException(
+            status_code=422,
+            detail="version mismatch for new product",
+        )
+    else:
```

Comment:
When you do an upfront check like above and you raise, you do not need to put an else, just put the code after the if.

Reply:
Oh okay, I'll keep this in mind next time. I think this code will be anyways replaced by the existing API code. So, no worries!